# This is my sample .ipynb file!
This could be your analyses files, your visualization files, etc.

In [1]:
import pandas as pd
import os
import numpy as np
import matplotlib.pyplot as plt
import pandas_datareader as pdr
import seaborn as sns
from tqdm import tqdm
import re

# Importing the Datasets


- WFH - % with access to a "flexible workplace"
- turnover - quit_rate
- productivity - % change in output per hour
- compensation - 12 months % change in wage

In [18]:
WFH = pd.read_csv('../stash-ka/inputs/employee_benefits.csv')
WFH.rename(columns = {'Series title ':'Industry'}, inplace = True)
WFH.rename(columns = {'Estimate':'WFH%'}, inplace = True)
WFH['WFH%'] = pd.to_numeric(WFH['WFH%'],errors='coerce')
WFH.info()
WFH.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 479 entries, 0 to 478
Data columns (total 24 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Series ID                    479 non-null    object 
 1   Industry                     479 non-null    object 
 2   Year                         479 non-null    int64  
 3   WFH%                         475 non-null    float64
 4   Estimate footnote            4 non-null      float64
 5   Standard error               479 non-null    float64
 6   Standard error footnote      0 non-null      float64
 7   Survey code                  479 non-null    object 
 8   Survey text                  479 non-null    object 
 9   Ownership code               479 non-null    int64  
 10  Ownership text               479 non-null    object 
 11  Industry code                479 non-null    object 
 12  Industry text                479 non-null    object 
 13  Occupation code     

Year        WFH%  Estimate footnote  Standard error   
count   479.000000  475.000000                4.0      479.000000  \
mean   2016.175365    7.551579                1.0        1.136326   
std       3.749273    6.827514                0.0        0.657954   
min    2010.000000    1.000000                1.0        0.100000   
25%    2013.000000    3.000000                1.0        0.650000   
50%    2016.000000    5.000000                1.0        1.000000   
75%    2019.000000    9.000000                1.0        1.400000   
max    2022.000000   39.000000                1.0        3.800000   

       Standard error footnote  Ownership code  Occupation code  Subcell code   
count                      0.0      479.000000            479.0         479.0  \
mean                       NaN        1.943633              0.0           0.0   
std                        NaN        0.632915              0.0           0.0   
min                        NaN        1.000000              0.0           0.0   
25%                        NaN        2.000000              0.0           0.0   
50%                        NaN        2.000000              0.0           0.0   
75%                        NaN        2.000000              0.0           0.0   
max                        NaN        3.000000              0.0           0.0   

       Estimate code  Datatype code  Provision code  
count          479.0          479.0           479.0  
mean            73.0           33.0           357.0  
std              0.0            0.0             0.0  
min             73.0           33.0           357.0  
25%             73.0           33.0           357.0  
50%             73.0           33.0           357.0  
75%             73.0           33.0           357.0  
max             73.0           33.0           357.0

In [ ]:
industry_name_list=[]

for industry in WFH['Industry']:
    industry_name = industry.split('in ')[1].capitalize()
    industry_name_list.append(industry_name)
    
WFH['Industry'] = industry_name_list

In [12]:
turnover = pd.read_csv('../stash-ka/inputs/turnover.csv')

In [17]:
turnover.rename(columns = {'Series Id':'Industry'}, inplace = True)
turnover.rename(columns = {'Value':'quit_rate'}, inplace = True)
turnover.info()
turnover.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Year       351 non-null    int64  
 1   Industry   351 non-null    object 
 2   quit_rate  351 non-null    float64
dtypes: float64(1), int64(1), object(1)
memory usage: 8.4+ KB


Year   quit_rate
count   351.000000  351.000000
mean   2016.000000    1.786895
std       3.746999    0.997711
min    2010.000000    0.300000
25%    2013.000000    1.100000
50%    2016.000000    1.600000
75%    2019.000000    2.300000
max    2022.000000    5.800000

In [20]:
productivity = pd.read_csv('../stash-ka/inputs/output_per_hour_pct.csv')
productivity.info()
productivity.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 209 entries, 0 to 208
Data columns (total 3 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Year                         209 non-null    int64  
 1   % change in output per hour  209 non-null    float64
 2   Industry                     209 non-null    object 
dtypes: float64(1), int64(1), object(1)
memory usage: 5.0+ KB


Year  % change in output per hour
count   209.00000                   209.000000
mean   2016.00000                     1.189952
std       3.16987                     2.891972
min    2011.00000                   -10.900000
25%    2013.00000                    -0.600000
50%    2016.00000                     1.100000
75%    2019.00000                     2.800000
max    2021.00000                    11.300000

In [22]:
compensation = pd.read_csv('../stash-ka/inputs/pct_change_compensation.csv')
compensation.rename(columns = {'12 months pct change in wage':'Industry'}, inplace = True)
compensation.rename(columns = {'Estimate Value':'% change in average wage'}, inplace = True)
compensation.info()
compensation.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260 entries, 0 to 259
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Year                      260 non-null    int64  
 1   Industry                  260 non-null    object 
 2   % change in average wage  260 non-null    float64
 3   Standard Error            260 non-null    float64
dtypes: float64(2), int64(1), object(1)
memory usage: 8.2+ KB


Year  % change in average wage  Standard Error
count   260.000000                260.000000      260.000000
mean   2016.000000                  2.711154        0.442308
std       3.748874                  1.194966        0.386723
min    2010.000000                  0.000000        0.100000
25%    2013.000000                  1.900000        0.200000
50%    2016.000000                  2.400000        0.400000
75%    2019.000000                  3.200000        0.500000
max    2022.000000                  8.100000        4.400000

# EDA
We envision a singular WFH dataset that encompasses the main variables from all of the imported datasets. More specifically, it would contain the following variables: 
 - `Year` (across all datasets)
 - `Industry` (across all datasets)
 - `WFH%` (WFH dataset)
 - `quit_rate` (Turnover dataset)
 - `% change in output per hour` (Productivity dataset)
 - `% change in average wage` (Compensation dataset)
 
Our final dataset will be in panel format, by industry and year (2010-2022). 

### WFH (3vars to keep)
This dataset, obtained from the BLS Employee Benefit Survey, contains the main WFH % variable for our analysis, it sets the stage for categorizing each industry. An initial problem that immediately pops out is the data type of the WFH variable, it must be changed to a numeric type to behave correctly in our analysis. Additionally, the excess columns must be dropped in order to simplify the merge. The highest WFH rate is 39% and the lowest is 1%, numbers we believe are quite feasible.

### Turnover (3vars to keep)
This dataset, obtained form the BLS JOLTS database, is very straightforward. It contains solely the industry, year, and quit_rate. After describing the variables, the minimum is 0.3% and the maximum is 5.8%. We did not encounter many difficulties when utilizing this dataset. 

### Productivity
This dataset, obtained from the BLS Office of Productivity and Technology, specifically references the `output per hour` variable that they track in their labor productivity/Total Factor Productivity databases. This variable is further broken down into industry and year, like the others. There was one issue that we found which may bring up problems; this dataset only spans into 2021, not 2022. This may influence our creation of visualizations in the future. All other variables seem sufficient. % change in output per hour max is 11.3 and the min is -10.9.

### Compensation
This dataset, obtained from the BLS Current Population Survey, provides the compensation variable for our study, `% change in average wage`. When importing the dataset initially, the coumns names are also labeled incorrectly so this needed to be fixed. The % change variable is of sufficient type and its maximum is 8.1% and the minimum is 0%. 